opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py

```diff
@@ -84,34 +117,54 @@ class MeterProvider(APIMeterProvider):
 
     def __init__(
         self,
+        *,
+        metric_readers: Sequence[MetricReader],
         resource: Resource = Resource.create({}),
+        views: Sequence[View] = (),
         shutdown_on_exit: bool = True,
     ):
         self._resource = resource
+        self._views = views
         self._atexit_handler = None
 
         if shutdown_on_exit:
             self._atexit_handler = register(self.shutdown)
 
-        self._metric_readers = []
+        self._metric_readers = metric_readers
         self._metric_exporters = []
-        self._views = []
         self._shutdown = False
 
+        self._measurement_processor = DefaultMeasurementConsumer(
```

Comment:
Should all `MetricReader`s use the same `MeasurementConsumer` instance? Looks kind of strange that `MeasurementConsumer` has a reference to the list of metric readers and for each metric reader, in the following lines, the callback is set for the one instances collect method. I would assume each metric reader is the start of a collection "pipeline", so should have different instances of Measurement Consumer each?

Reply:
Possible use case for a "multi" measurement processor.